firefox-ios/Client/Ecosia/Extensions/SearchViewController+Ecosia.swift

```diff
@@ -0,0 +1,133 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import UIKit
+import Common
+import Ecosia
+
+// MARK: - AI Search Autocomplete Extensions
+extension SearchViewController {
+
+    // MARK: - Helper Methods
+
+    /// Check if current row is the AI Search item
+    func isAISearchRow(_ indexPath: IndexPath) -> Bool {
+        guard SearchListSection(rawValue: indexPath.section) == .searchSuggestions else { return false }
+        let shouldShowAISearch = AISearchMVPExperiment.isEnabled && !viewModel.searchQuery.isEmpty
+        guard shouldShowAISearch else { return false }
+
+        let count = viewModel.suggestions?.count ?? 0
+        let maxSuggestions = count < 4 ? count : 4
+        return indexPath.row == maxSuggestions
+    }
+
+    /// Calculate number of rows including AI Search item if enabled
+    func numberOfRowsForSearchSuggestions() -> Int {
+        guard let count = viewModel.suggestions?.count else { return 0 }
+        let maxSuggestions = count < 4 ? count : 4
```

Comment:
Is `4` because that's the most Firefox ever shows us? Would be nice to extract this into a const so it's clearer.

Reply:
Also, using `min` like done of the Safe array access reads better I believe.

Reply:
Yeah, no worries, might extract into a const anyway since we have it in multiple places, maybe we should use it there on Firefox too to make sure this doesn't break anything then.